packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2626,6 +2626,8 @@ export class ContainerRuntime
 		await this.pendingStateManager.applyStashedOpsAt(message.sequenceNumber);
 	}
 
+	private noMoreLoggingLegacyCase = false;
```

Comment:
Consider creating a closure function which maintains it's own `alreadyInvokedOnce` state. 
```
logLegacyCase = (function() {
    let alreadyInvokedOnce = false;
    
    return () => {
        if (!alreadyInvokedOnce) {
            alreadyInvokedOnce = true;
           // log here
        }
    };
)();
```

Reply:
That's how I started, but that function would have to be defined at the outer scope to have the desired effect (as-is, the function is defined anew on every call to `process`).  But the function already captures the message type into its closure, so then I would need to make it a curried function (take the type, return a function that takes the codepath), and I thought this was simpler.  Thoughts on that tradeoff?

Reply:
Ok there I did it.  Slightly different approach where we just clear the function definition when it's invoked, it's more concise.  Downside is the function variable is declared with `let` so someone could overwrite it but that's not going to happen accidentally.